src/mpi/coll/src/coll_impl.c

```diff
@@ -52,32 +52,78 @@ int MPIR_COLL_init(void) {
     MPIR_Allgather_alg_inter_choice = MPIR_ALLGATHER_ALG_INTER_AUTO;
 
     /* Allgatherv Intra */
+    if (0 == strcmp(MPIR_CVAR_ALLGATHERV_ALGORITHM_INTRA, "brucks"))
+        MPIR_Allgatherv_alg_intra_choice = MPIR_ALLGATHERV_ALG_INTRA_BRUCKS;
+    else if (0 == strcmp(MPIR_CVAR_ALLGATHERV_ALGORITHM_INTRA, "recursive_doubling"))
+        MPIR_Allgatherv_alg_intra_choice = MPIR_ALLGATHERV_ALG_INTRA_RECURSIVE_DOUBLING;
+    else if (0 == strcmp(MPIR_CVAR_ALLGATHERV_ALGORITHM_INTRA, "ring"))
+        MPIR_Allgatherv_alg_intra_choice = MPIR_ALLGATHERV_ALG_INTRA_RING;
+    else
+        MPIR_Allgatherv_alg_intra_choice = MPIR_ALLGATHERV_ALG_INTRA_AUTO;
 
     /* Allgatherv Inter */
+    MPIR_Allgatherv_alg_inter_choice = MPIR_ALLGATHERV_ALG_INTER_AUTO;
 
     /* Allreduce_group Intra */
 
     /* Allreduce_group Inter */
```

Comment:
I know some of the collectives don't have alternative implementations (allred_group, exscan, gatherv, neighborhood collectives, etc.), but I think everything should get a CVAR.

Reply:
If we're not going to do allreduce_group (and any others that we're not going to do, let's take them out of this list.

Reply:
@wesbland In Allgatherv Inter above, As there is only one option we have not used string comparison. Should this also be changed?

Reply:
Yes, sorry I missed that one.

Reply:
@wesbland Ok, I have a to add a good number of changes then. Will do and update Thank you